CODE_REVIEW_CHECKLIST.md

```diff
@@ -69,3 +71,10 @@ may have.
 
 - [ ] Will this change work on all of our supported platforms and
       browsers?
+
+- [ ] Do we expect significant changes in performance due to the
```

Comment:
You're right about this leading into long, dark alleys... My intention with including this was merely to have a reminder -- _"Is this going to be trouble for big deployments?"_ (where "big" again is very vague)...

Trying to come up with recent changes that would have been relevant for this rule, we've had the added functional PG index on `lower(email)` to allow for faster case-insensitive lookups. Adding that index feels instantaneous on my small users table, but could have been an issue for Hosted Chef. For this reason, we had changed the code so it'll create the index asynchronously.

Now this example is only for "upgrade-time", i.e. how long does it take to upgrade your chef-server, but I assume similar things could exist for others. (In the end, adding the functional index is a reaction to changing the search query -- only changing the query without adding the index would have turned a fast PG index lookup into a slow sequential scan.)

So, I don't believe this is a clear-cut criterium for accepting a change, but rather a reminder -- I could see value in splitting the list into two sections, but I also don't want to introduce too much possibly unnecessary structure -- what would you suggest, @blakestier? 😃